.gitignore

```diff
@@ -3,4 +3,3 @@ build
 coverage
 .nyc*
 .idea
-package-lock.json
```

Comment:
Neither `package-lock.json` nor `npm-shrinkwrap.json` should not be added to "libraries" or "components" like livepush, only to "final applications" like the balena CLI or the balena supervisor. Reasons:
* `package-lock.json` is not publishable to the npm registry -- meaning that, if someone runs `npm install livepush` for the their application, the `package-lock.json` of livepush will be ignored. As a result, any dependencies pinned by the `package-lock.json` file of livepush will be unpinned at the point of usage (balena CLI, balena supervisor), and any bugs or issues resulting from unpinned dependencies will be visible at the point of usage and hidden / masked in the livepush repo where tests are run and where issues should be detected.
> https://docs.npmjs.com/cli/v6/configuring-npm/package-lock-json  
> _"One key detail about package-lock.json is that it cannot be published, and it will be ignored if found in any place other than the toplevel package. It shares a format with npm-shrinkwrap.json, which is essentially the same file, but allows publication. This is not recommended unless deploying a CLI tool or otherwise using the publication process for producing production packages."_

* `npm-shrinkwrap.json` in a component like `livepush` can be published and will be honoured downstream, but almost always in a harmful way. It prevents the downstream applications (like the balena CLI or the balena supervisor) from excluding / updating transitive dependencies (e.g. a dependency of livepush, that may be partially shared with dependencies of other modules) if a security issue or bug is found, and it results in bloated `node_modules` folders with unnecessarily duplicate transitive dependencies. I have found myself submitting PRs to 3rd party balena CLI dependency modules (components) to remove the `npm-shrinkrap.json`, which was resulting in 200MB extra disk space in the CLI's `node_modules` folder if I recall correctly.
> https://docs.npmjs.com/cli/v6/configuring-npm/shrinkwrap-json
> _"The recommended use-case for npm-shrinkwrap.json is applications deployed through the publishing process on the registry: for example, daemons and command-line tools intended as global installs or devDependencies. It's strongly discouraged for library authors to publish this file, since that would prevent end users from having control over transitive dependency updates."_

So what can you do to pin dependencies in `livepush`? You can only `package.json`, which usually pins major versions only. What if a bug or malware is suddenly introduced in a minor / patch dependency of livepush? In this case, downstreams applications (CLI, supervisor) will not be immediately affected because they are using their own `npm-shrinkwrap.json` file. In this rare situation, `livepush` can only pin the minor / patch release using `package.json`.

So my review suggestion for this PR is that it be closed. :-)

Reply:
ahhh. Thank you for explaining that.